### PR TITLE
chore: Use string to access SQLite columns

### LINKED ISSUE #1531: ENH: Replace numeric indexing of SQLite columns with column names
### Is your feature request related to a problem?

Currently, column indexing is prone to issues with off by one, copy paste and ambiguity.  This has caused issues in the past and has caused issues to this day and will cause issues in the future if a column is added to the query or removed in an update.

### Describe the solution you'd like

Replace any calls of, for example a query with a column `banana` and a getter `getIntField(0)` with the getter `getIntField("banana")`.  there are other similar api calls which should have this done as well.

### Repository breaking implications

Binary size will grow but we're talking a few hundred bytes.  Worth the zero error and runtime error if a column doesnt exist in an updated query.

### Describe alternatives you've considered

Why are these not just using the ORM in these random queries?  this should be done in the future.  No file should need to link with sqlite.

### Additional context

i ate a carrot today.  tasted good.

### PR DESCRIPTION
**Pull request by EmosewaMC**
~~Untested right now.  Will try to test later.  May do some other small improvements near this code, like removing finalizes and some char* -> string -> copy to data stuff, may not.  idk atm.~~

fixes #1531

### CODE REVIEW

**Review by Wincent01 — COMMENTED** [review]
It probably wouldn't be a measurable improvement, but we could utilize static variables to store a cashed index of the fields.

**Comment by EmosewaMC** [comment]
I was planning on doing more improvements in a separate PR.  this one is mostly just a number->text replacement and ideally any queries to the cdclient are replaced with an ORM table for caching

**Comment by EmosewaMC** [comment]
tested the following
combat in nimbus station with a stromling and nuckal is working and effects by the stromling play
lookup works
having a pirate hat on and transferring to another zone works (nimbus station ->gnarled forest)
set bonuses work (used 6 piece engineer with valiants)
buffs work(equipped ninjago chest and pants and reduced damage by 2)
projectiles still land and do damage
character create still chooses the correct shirts the player chose
switch multiple works (tested with valiant summoner weapon chargeup)
claiming and placing a model works
paradox teleporter in nexus tower has the correct animation time
adding the nuckal mission and smashing them gives the quest item and does not when you do not have the mission
can use mounts still
can dismantle models to bricks in the inventory

**Comment by EmosewaMC** [comment]
lookup still works

**Review by aronwk-aaron — DISMISSED** [review]

File: dGame/dInventory/ItemSet.cpp [hunk 1]
```diff
@@ -27,14 +27,16 @@ ItemSet::ItemSet(const uint32_t id, InventoryComponent* inventoryComponent) {
 		return;
 	}
 
-	for (auto i = 0; i < 5; ++i) {
-		if (result.fieldIsNull(i)) {
+	const std::vector<std::string_view> rowNames = { "skillSetWith2", "skillSetWith3", "skillSetWith4", "skillSetWith5", "skillSetWith6" };
```
Comment (jadebenn) [inline dGame/dInventory/ItemSet.cpp:30]: nit: could this be a constexpr std::array?

File: dGame/dInventory/ItemSet.cpp [hunk 2]
```diff
@@ -27,14 +27,16 @@ ItemSet::ItemSet(const uint32_t id, InventoryComponent* inventoryComponent) {
...
+	for (auto i = 0; i < rowNames.size(); ++i) {
```
Comment (jadebenn) [inline dGame/dInventory/ItemSet.cpp:31]: nit: could this be a range-for loop?

**Review by jadebenn — DISMISSED** [review]
nitpicks aside, looks good. you can address them if you want, or don't if you don't

File: dGame/dInventory/ItemSet.cpp (hunk 2 above)
Comment (jadebenn) [reply dGame/dInventory/ItemSet.cpp:33]: Ah, never-mind. I see why it can't now.

File: dGame/dInventory/ItemSet.cpp (hunk 1 above)
Comment (jadebenn) [reply dGame/dInventory/ItemSet.cpp]: Changed it to be one.

**Review by jadebenn — APPROVED** [review]

**Comment by jadebenn** [comment]
well, frick

**Comment by jadebenn** [comment]
welp, reverting my attempted changes and approving. that is a mystery to solve later.

**Review by jadebenn — APPROVED** [review]

**Review by jadebenn — APPROVED** [review]

File: dGame/dInventory/ItemSet.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dGame/dInventory/ItemSet.cpp]: I would appreciate not editing my PRs unless I have asked someone to, or worked together with someone on it.  I did not think those changes were necessary and would have left them alone personally as they would not change anything.

**Review by aronwk-aaron — APPROVED** [review]

File: dGame/dInventory/ItemSet.cpp (hunk 1 above)
Comment (jadebenn) [reply dGame/dInventory/ItemSet.cpp]: I was not aware of that. Will do moving forward.

**Pull request merged**